src/@next/components/atoms/I18nLoader/I18nLoader.tsx

```diff
@@ -0,0 +1,45 @@
+import React from "react";
+import { I18nContext } from ".";
+import { IProps } from "./types";
+
+const fallbackLanguage = "en";
+
+const getLangCode = (code: string) =>
+  code.includes("-") ? code.split("-")[0] : code;
+
+const defaultLanguage = getLangCode(
+  (
+    (navigator.languages && navigator.languages[0]) ||
+    navigator.language ||
+    fallbackLanguage
+  ).toLowerCase()
+);
+
+export const I18nLoader: React.FC<IProps> = ({ children, languages }: IProps) => {
+  const [ language, setLanguage ] = React.useState<string>(() => {
+    return languages.hasOwnProperty(defaultLanguage) ? defaultLanguage : fallbackLanguage
+  });
+  const [ catalogs, setCatalogs ] = React.useState<any>({});
+
+  React.useEffect(() => {
+    const loadCatalog = async (language:string) => {
```

Comment:
I think it might be better to move loadCatalog method outside useEffect and return catalog from it:
`const catalog = loadCatalog(language)`